tracpro/profiles/__init__.py

```diff
@@ -52,18 +53,31 @@ def _user_get_full_name(user):
     """
     Override regular get_full_name which returns first_name + last_name
     """
-    return user.profile.full_name if user.has_profile() else " ".join([user.first_name, user.last_name]).strip()
+    if user.has_profile():
+        return user.profile.full_name
+    return super(User, user).get_full_name()
 
 
-def _user_get_regions(user, org):
+def _user_get_direct_regions(user, org):
+    """Return org regions user has direct permission for."""
     def calculate():
         # org admins have implicit access to all regions
         if user.is_admin_for(org):
             return Region.get_all(org)
         else:
             return user.regions.filter(is_active=True)
+    attr_name = '_regions_{}'.format(org.pk)  # cache per org
+    return get_obj_cacheable(user, attr_name, calculate)
 
-    return get_obj_cacheable(user, '_regions', calculate)
+
+def _user_get_all_regions(user, org):
+    """Return org regions user has direct or implied (by hierarchy) permission for."""
+    def calculate():
+        regions = user.get_direct_regions(org).get_descendants(include_self=True)
+        regions = regions.filter(is_active=True)
+        return regions
+    attr_name = '_regions_with_descendants_{}'.format(org.pk)  # cache per org
+    return get_obj_cacheable(user, attr_name, calculate)
```

Comment:
I'm all for making granular, re-usable methods but I'm wondering why you split out get_direct_regions and get_all_regions because get_all_regions() is the only one that seems to be called.